src/utils/NewsApi.js

```diff
@@ -0,0 +1,43 @@
+import { fromDate, toDate } from "./helpers";
+
+const BASE_URL = "https://nomoreparties.co/news/v2/everything?";
+const API_KEY = "397db29f2b7d47238daa4bdcefcbdfd2";
+
+export class NewsApi {
+  constructor(baseUrl, apiKey, fromDate, toDate) {
+    this.apiKey = apiKey;
+    this.baseUrl = baseUrl;
+    this.date = { from: fromDate, to: toDate };
+  }
+
+  getArticleData = (query) => {
+    return fetch(
+      `${this.baseUrl}q=${query}&language=en&from=${this.date.from}&to=${this.date.to}&pageSize=100&apiKey=${this.apiKey}`
+    )
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          return Promise.reject(`error :${res.message}`);
+        }
+      })
+      .then((res) => {
+        return res.articles.map((item) => {
+          const {
+            source: { name: source },
+            title,
+            publishedAt: date,
+            description: text,
+            url: link,
+            urlToImage: image,
+          } = item;
+          return { title, source, date, text, link, image };
+        });
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `NewsApi` (or any Api) because you need to `catch` possible errors at the very end of any request and this end is in `index.js` (`App.js`)  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.